ArduSub/mode_poshold.cpp

```diff
@@ -103,12 +83,69 @@ void ModePoshold::run()
             attitude_control->input_euler_angle_roll_pitch_euler_rate_yaw(target_roll, target_pitch, target_yaw_rate);
             sub.last_pilot_heading = ahrs.yaw_sensor; // update heading to hold
 
-        } else { // call attitude controller holding absolute absolute bearing
+        } else { // call attitude controller holding absolute bearing
             attitude_control->input_euler_angle_roll_pitch_yaw(target_roll, target_pitch, sub.last_pilot_heading, true);
         }
     }
 
-    // Update z axis //
+    // update z axis
     control_depth();
+
+    // update xy axis
+    // call this after Sub::get_pilot_desired_climb_rate is called so that THR_DZ is reasonable
+    control_horizontal();
+}
+
+// behavior is similar to Sub::get_pilot_desired_climb_rate
+float Sub::get_pilot_desired_horizontal_rate(RC_Channel *channel) const
```

Comment:
wouldn't it make sense to keep it alongside `get_pilot_desired_climbrate()` in attitude.cpp ?

Reply:
Good point, I'll move it.